src/components/structures/HostingSignupAction.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2020 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as React from "react";
+import Modal from "../../Modal";
+import HostingSignupDialog from "../views/dialogs/HostingSignupDialog";
+import {
+    IconizedContextMenuOption,
+    IconizedContextMenuOptionList,
+} from "../views/context_menus/IconizedContextMenu";
+import { _t } from "../../languageHandler";
+
+interface IProps {}
+
+interface IState {}
+
+export default class HostingSignupAction extends React.PureComponent<IProps, IState> {
+    closingAllowed = false;
+    modalRef: any;
```

Comment:
this should be typed, and if it can't be then it needs a comment explaining why.